library/modules/Gui.cpp

```diff
@@ -2911,56 +2923,91 @@ bool Gui::inRenameBuilding()
     return false;
 }
 
-bool Gui::getViewCoords (int32_t &x, int32_t &y, int32_t &z)
+bool Gui::getViewCoords(int32_t &x, int32_t &y, int32_t &z)
 {
     x = *df::global::window_x;
     y = *df::global::window_y;
     z = *df::global::window_z;
     return true;
 }
 
-bool Gui::setViewCoords (const int32_t x, const int32_t y, const int32_t z)
+bool Gui::setViewCoords(const int32_t x, const int32_t y, const int32_t z)
 {
     (*df::global::window_x) = x;
     (*df::global::window_y) = y;
     (*df::global::window_z) = z;
     return true;
 }
 
-bool Gui::getCursorCoords (int32_t &x, int32_t &y, int32_t &z)
+bool Gui::getCursorCoords(int32_t &x, int32_t &y, int32_t &z)
 {
-    x = df::global::cursor->x;
-    y = df::global::cursor->y;
-    z = df::global::cursor->z;
+    using df::global::cursor;
+    bool is_adv = World::isAdventureMode();
+    if (is_adv || !cursor)
+    {
+        df::coord p;
+        if (is_adv && game)
+        {
+            auto &look = game->main_interface.adventure.look;
+            if (look.open)
+                p = look.cursor;
+        }
+        x = p.x; y = p.y; z = p.z;
+        return p.isValid();
+    }
+
+    x = cursor->x; y = cursor->y; z = cursor->z;
     return has_cursor();
 }
 
-bool Gui::getCursorCoords (df::coord &pos)
+bool Gui::getCursorCoords(df::coord &pos)
 {
-    pos.x = df::global::cursor->x;
-    pos.y = df::global::cursor->y;
-    pos.z = df::global::cursor->z;
-    return has_cursor();
+    using df::global::cursor;
+    df::coord p;
+    if (World::isAdventureMode())
+    {
+        if (game)
+        {
+            auto &look = game->main_interface.adventure.look;
+            if (look.open)
+                p = look.cursor;
+        }
+    }
+    else if (cursor)
+        p = df::coord(cursor->x, cursor->y, cursor->z);
+
+    pos = p;
+    return p.isValid();
```

Comment:
```suggestion
    int32_t x, y, z;
    bool ret = getCursorCoords(x, y, z);
    pos.x = x;
    pos.y = y;
    pos.z = z;
    return ret;
```

Reply:
to avoid code duplication?

Reply:
I think `getCursorPos` works better, since the `int32_t` is lost anyway.